examples/DivisorCurve.jl

```diff
@@ -0,0 +1,313 @@
+
+Given a curve `C` which is assumed to be smooth and irreducible, return the divisor on the curve `C` defined by `D`.
+"""
+struct ProjCurveDivisor{S <: FieldElem} <: CurveDivisor
+    C::ProjPlaneCurve{S}
+    divisor::Dict{Oscar.Geometry.ProjSpcElem{S}, Int}
+    degree::Int
+    function ProjCurveDivisor(C::ProjPlaneCurve{S}, D::Dict{Oscar.Geometry.ProjSpcElem{S}, Int}) where S <: FieldElem
+        for (P, m) in D
+            P in C || error("The point ", P, " is not on the curve")
+            if m == 0
+                delete!(D, P)
+            end
+        end
+        new{S}(C, D, sum(values(D)))
+    end
+end
+
+function ProjCurveDivisor(C::ProjPlaneCurve{S}, P::Oscar.Geometry.ProjSpcElem{S}, m::Int=1) where S <: FieldElem
+    return ProjCurveDivisor(C, Dict(P => m))
+end
+
+################################################################################
+# To get a nice output for the divisors. The needs_parentheses is needed since
+# expressify wants to know if parentheses are needed.
+
+function AbstractAlgebra.needs_parentheses(A::Array{T, 1}) where T <: RingElem
+    return false
+end
+
+function AbstractAlgebra.expressify(@nospecialize(a::AffineCurveDivisor); context = nothing)
+   prod = Expr(:call, :+)
+   for (P, m) in a.divisor
+      ep = AbstractAlgebra.expressify(P.coord, context = context)
+      push!(prod.args, Expr(:call, :*, m, ep))
+   end
+   return prod
+end
+
+function AbstractAlgebra.needs_parentheses(A::Oscar.Geometry.ProjSpcElem{T}) where T <: RingElem
+    return false
+end
+
+function AbstractAlgebra.expressify(@nospecialize(a::ProjCurveDivisor); context = nothing)
+   prod = Expr(:call, :+)
+   for (P, m) in a.divisor
+      ep = AbstractAlgebra.expressify(P, context = context)
+      push!(prod.args, Expr(:call, :*, m, ep))
+   end
+   return prod
+end
+
+function Base.show(io::IO, ::MIME"text/plain", a::CurveDivisor)
+  print(io, AbstractAlgebra.obj_to_string(a, context = io))
+end
+
+function Base.show(io::IO, a::CurveDivisor)
+  print(io, AbstractAlgebra.obj_to_string(a, context = io))
+end
+
+################################################################################
+################################################################################
+
+function curve(D::CurveDivisor)
+    return D.C
+end
+
+function Oscar.degree(D::CurveDivisor)
+    return D.degree
+end
+
+################################################################################
+
+function _check_same_curve(D::CurveDivisor, E::CurveDivisor)
+    D.C == E.C || error("The divisors are not on the same curve")
+end
+
+################################################################################
+################################################################################
+# Sum of two divisors
+
+function Base.:+(D::AffineCurveDivisor, E::AffineCurveDivisor)
+    _check_same_curve(D, E)
+    return AffineCurveDivisor(D.C, mergewith(+, D.divisor, E.divisor))
```

Comment:
Actually, I mixed things up: we support Julia >= 1.3 but `mergewith` requires Julia >= 1.5. However, the same functionality is available under the name `merge` in all Julia versions; they merely deprecated it in Julia 1.5 in favor of the name `mergewith`. So for now, we can do this:
```suggestion
    return AffineCurveDivisor(D.C, merge(+, D.divisor, E.divisor))
```